sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/TriggerRunner.java

```diff
@@ -225,9 +225,24 @@ public void clearState(W window, Timers timers, StateAccessor<?> state) throws E
     rootTrigger.invokeClear(contextFactory.base(window, timers, rootTrigger, finishedSet));
   }
 
+  /**
+   * Return {@literal true} if trigger or any of its children has any non-empty state, including
+   * any finished bits.
+   */
+  public boolean hasState(W window, Timers timers, StateAccessor<?> state) throws Exception {
```

Comment:
Now just check for whether the trigger has any finished bits. Can't just check if it is finished since the Repeatably trigger is never finished yet does not have finished bits.